core/src/main/scala/filodb.core/store/ChunkSetInfo.scala

```diff
@@ -244,34 +250,36 @@ trait ChunkInfoIterator { base: ChunkInfoIterator =>
 
 object ChunkInfoIterator {
   val empty = new ChunkInfoIterator {
+    def close(): Unit = {}
     def hasNext: Boolean = false
     def nextInfo: ChunkSetInfo = ChunkSetInfo(0)
   }
-
-  def single(info: ChunkSetInfo): ChunkInfoIterator = new ChunkInfoIterator {
-    var count = 0
-    def hasNext: Boolean = count == 0
-    def nextInfo: ChunkSetInfo = { count += 1; info }
-  }
 }
 
 class ElementChunkInfoIterator(elIt: ElementIterator) extends ChunkInfoIterator {
+  def close(): Unit = elIt.close()
   def hasNext: Boolean = elIt.hasNext
```

Comment:
In general yes, although the implementation isn't expected to throw any exceptions.

Reply:
I realize that this may get ugly, but as I mentioned in another thread of this PR, exceptions can be thrown by upstream query processing logic. We may need to handle exceptions on next/hasNext at the topmost level and propagate the close() call down.